lib/jose-jwe-utils.js

```diff
@@ -26,34 +26,85 @@ var Utils = {};
  *
  * @param rsa_key  public RSA key in json format. Parameters can be base64
  *                 encoded, strings or number (for 'e').
+ * @param alg      String, name of the algorithm
  * @return Promise<CryptoKey>
  */
-JoseJWE.Utils.importRsaPublicKey = function(rsa_key) {
-  var jwk = Utils.convertRsaKey(rsa_key, ["n", "e"]);
-  var config = getCryptoConfig("RSA-OAEP");
-  return crypto.subtle.importKey("jwk", jwk, config.id, false, ["wrapKey"]);
+Jose.Utils.importRsaPublicKey = function (rsa_key, alg) {
+  var jwk,
+    config = null,
+    rk,
+    usage;
+
+  alg = alg || rsa_key.alg || "RSA-OAEP";
```

Comment:
This kind of code can lead to bugs down the road. I would remove "RSA-OAEP" as the default algorithm. I would also assert that if rsa_key.alg is set, it must match alg instead of picking whichever is set.